packages/esm-patient-test-results-app/src/overview/common-overview.component.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { useCallback } from 'react';
+import Table16 from '@carbon/icons-react/es/table/16';
+import ChartLine16 from '@carbon/icons-react/es/chart--line/16';
+import { Button, TableToolbarContent, TableToolbar } from 'carbon-components-react';
+import { EmptyState } from '@openmrs/esm-patient-common-lib';
+import { headers, formatDate, InfoButton, Separator } from './helpers';
+import { OverviewPanelEntry } from './useOverviewData';
+import { useTranslation } from 'react-i18next';
+import { navigate } from '@openmrs/esm-framework';
+import CommonDataTable from './common-datatable.component';
+import styles from './common-overview.scss';
+
+const DashboardResultsCount = 5;
+
+interface CommonOverviewPropsBase {
+  overviewData: Array<OverviewPanelEntry>;
+  insertSeparator?: boolean;
+  patientUuid?: string;
+}
+
+interface CommonOverviewPropsWithToolbar {
+  openTimeline: (panelUuid: string) => void;
+  openTrendline: (panelUuid: string, testUuid: string) => void;
+}
+
+interface CommonOverviewPropsWithoutToolbar {
+  hideToolbar: true;
+}
+
+type Only<T, U> = {
+  [P in keyof T]: T[P];
+} & {
+  [P in keyof U]?: never;
+};
+
+type Either<T, U> = Only<T, U> | Only<U, T>;
```

Comment:
This doesn't work if `T` contains any properties that `U` contains: [TS Playground #1](https://www.typescriptlang.org/play?#code/C4TwDgpgBA8gdgGxAHgCoBooFUB8UC8UA3gFBRQDaAClAJZxQDWEIA9gGZSoC6AXF9W4BuEgF8oAMmJlKNekxYds3APz84EAG4QATiNEiSoSFACitYAAtdaTLgKxEKDNjwAfR0mRZMqHIfpgXXYAQwBjaABBaXIQ-gAjVlYECBC4GXiEpJS0sRISQODw6AAhGKhMiuzU9PIwrOSavKNwKNYdMsJzKxtIzBL-fLDWOABnYCh2JP5I9s7iKDioYB0AVwhMStCEUehREiA)

We can fix it by adding an Omit, see [TS Playground #2](https://www.typescriptlang.org/play?#code/C4TwDgpgBA8gdgGxAHgCoBooFUB8UC8UA3gFBRQDaAClAJZxQDWEIA9gGZSoC6AXF9W4BuEgF8oAMmJlKNekxYdYAW1rBkWTMzadUObgH5+cCADcIAJxGiRJUJCgBRNQAtLaTLgKxEKDNjwAHx8kDUw9W3pgS3YAQwBjaABBaXJY-gAjVlYECFi4GQzM7Nz8sRISKJiE6AAhVKgixpK8gvJ44pzW8rtwZNYLesJnYDcLZCTMWpxbeNY4AGdgKHSoJIGh4hX+YAsAVwhMJriEBehREjnF5ab1we8iRp39w6gOqBOzqAurpcbYgAmAEZ+HdNo9VrsDt9LvM-hlAQAmUEbB5PKBQ86w67-AEAZhR90Ij3emJhvxugIALITwdsMS8js8DphSS8YUA)

```suggestion
// Defining a disjoint union type `Either`
type Only<T, U> = {
  [P in keyof T]: T[P];
} & {
  [P in keyof Omit<U, keyof T>]?: never;
};

type Either<T, U> = Only<T, U> | Only<U, T>;
```

Reply:
I know it doesn't matter for this particular use case, but a good Disjoint Union type is a nice thing to have around :)